File: 2016/s1.py
# code by savir singh

def checker(a, b):
    alist = list(a)
    alist.sort()
    blist = list(b)
    blist.sort()

    return (alist == blist)

n = input()
anagram = input()
counter = 0
asterisks = anagram.count("*")

ana2 = list(anagram)
n2 = list(n)

if "*" not in anagram:
    if checker(n, anagram):
        print("A")
    else:
        print("N")

else:
    ana1 = list(anagram)
    while '*' in ana1:
        ana1.remove('*')
    yescount = 0
    for j in range(len(ana1)):
        if str(ana1[j]) in n:
            yescount += 1
        else:
            break
    if yescount == len(ana1) and len(ana1) + anagram.count("*") == len(n):
        print("A")
    else:
        print("N")

File: 2016/s1.cpp
// code by savir singh
// 9/15 - WA

#include <bits/stdc++.h>
#define ll long long
using namespace std;

int main() {
    string a, b;
    cin >> a >> b;
    string output = "A";
    bool stop=false;
    int ast = count(b.begin(), b.end(), '*');
    for (int i=0; i<a.length(); i++) {
        if (count(a.begin(), a.end(), a[i]) != count(b.begin(), b.end(), a[i])) {
            for (int j=1; j<=ast; j++) {
                if (count(a.begin(), a.end(), a[i])==count(b.begin(), b.end(), a[i])+j) {
                    output = "A";
                    ast-=j;
                    stop = true;
                    break;
                }
                else {
                    output = "N";
                }
            }
            if (stop==false) {
                output = "N";
                break;
            }
        }
    }
    cout << output;
}
